### PR TITLE
Open PriceForm in a Dialog

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Open PriceForm in a Dialog

## Justify why they are needed
Trying out the concept from the prototype of opening the form in a modal

https://www.figma.com/proto/LvU4ntV5hpDTdueWbafpro/Calculator-Prototype?page-id=0%3A1&node-id=4%3A5471&starting-point-node-id=4%3A4744

https://user-images.githubusercontent.com/6661511/199452580-0203eae3-e8cf-44ca-81c0-0003d5daf9eb.mov

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #794** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/794" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/794?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PriceCalculator/PriceCalculator.tsx [hunk 1]
```diff
@@ -31,6 +34,7 @@ export const PriceCalculator = ({ title }: Props) => {
   const handleSuccess = (updatedPriceIntent: PriceIntentFragmentFragment) => {
     setProductOffer(updatedPriceIntent.offers[0])
     refreshData()
+    setOpenDialog(false)
```
Comment (gustaveen) [inline apps/store/src/components/ProductPage/PriceCalculator/PriceCalculator.tsx:37]: Closes the Modal on success

File: apps/store/src/components/ProductPage/PriceCalculator/PriceCalculator.tsx [hunk 2]
```diff
@@ -91,15 +97,38 @@ export const PriceCalculator = ({ title }: Props) => {
...
-        <Space y={1}>
-          <PriceForm
-            form={form}
-            priceIntent={priceIntent}
-            onSuccess={handleSuccess}
-            onUpdated={refreshData}
-            loading={isLoadingData}
-          />
-        </Space>
+        <Dialog.Root open={openDialog} onOpenChange={setOpenDialog}>
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PriceCalculator/PriceCalculator.tsx:100]: Nit: this is large JSX block, can we extract a component here?  Maybe something local to this particular module

File: apps/store/src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -32,6 +32,7 @@ export const Window = styled.div(({ theme }) => ({
 const StyledContentWrapper = styled.div({
   position: 'fixed',
   inset: 0,
+  zIndex: 2000,
```
Comment (alebedev) [inline apps/store/src/components/Dialog/Dialog.tsx:35]: Nit: does it make sense to collect all z-indexes we use in one place?

**Review by alebedev — APPROVED** [review]
Looks good, some nit comments inside

File: apps/store/src/components/Dialog/Dialog.tsx (hunk 3 above)
Comment (gustaveen) [reply apps/store/src/components/Dialog/Dialog.tsx:35]: For sure! Coming up 🙌

**Pull request merged**